content/articles/creating-simple-visual-augmented-reality-application-in-android/index.md

```diff
@@ -0,0 +1,166 @@
+
+Add the highlighted classpath as shown here 
+
+![alt text](/engineering-education/creating-simple-visual-augmented-reality-application-in-android/dependence.png)
+
+Lastly, add the plugin id in build.gridle(:app)
+
+ ![alt text](/engineering-education/creating-simple-visual-augmented-reality-application-in-android/dependenciesid.png)
+
+
+
+### Adding permissions for camera
+
+On the AndroidManifest.xml page, add the following codes;
+
+```java    
+    <uses-permission android:name="android.permission.CAMERA"/>
+    <uses-feature android:name="android.hardware.camera.ar" android:required="true"/>
+```
+
+They should be as shown here
+ ![alt text](/engineering-education/creating-simple-visual-augmented-reality-application-in-android/permission.png)
+
+### Application design
+
+We now need to configure the layout of the application. On the activity_main.xml window under layout folder modify existing codes to these shown below.
+
+![alt text](/engineering-education/creating-simple-visual-augmented-reality-application-in-android/mainactivity.png)
+
+Note that we change the layout from the dafault layout to relative layout.
+
+### Java codes
+These codes are written in the MainActivity.Java window. These codes are:
+```java
+package com.example.AugmetedReality2;
+
+import androidx.appcompat.app.AppCompatActivity;
+import android.os.Bundle;
+
+public class MainActivity extends AppCompatActivity {
+    private ArFragmet arFragmet;
+
+       @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+
+        arFragmet=(ArFragmet)getSupportFragmentManager().findFragmentById(R.id.arFragment);
+        arFragmet.setOnTabArPlaneListener((hitResult, plane, motionEvent) ->{
+            final Anchor anchor=hitResult.createAnchor();
+            ModelRenderable.builder()
+                    .setSource(context:this Uri.parse("AR.sfb"))
+                    .build()
+                    .theAccept(modelRenderable -> addModelToScene(anchor, modelRenderable));
+
+        });
+        private void addModelToScene(Anchor anchor, ModelRenderable modelRenderable){
+            AnchorNode_node=new AnchorNode(anchor);
+            TrasformableNode trasformableNode =new TrasformableNode(arFragmet.getTransformationSystem());
+            trasformableNode.setParent(node);
+            trasformableNode.setRenderable(modelRenderable);
+
+
+            arFragmet.getArSceneView().getScene().addChild(node);
+            trasformableNode.select();
+        }
+    }
+}
+```
+This code should be implemented as shown:
+
+ ![alt text](/engineering-education/creating-simple-visual-augmented-reality-application-in-android/mainactivity_app.png)
+
+  ![alt text](/engineering-education/creating-simple-visual-augmented-reality-application-in-android/mainactivityjava.png)
+
+The codes will allow users to drag or trasform the 3D object.
+
+### Running the app
```

Comment:
In this step, I want you to have a gif that illustrates the end product of the app. I also want you to have the GitHub repository which will host the code for this project linked here. Also, the downloadable sample apk which can be hosted on Google drive.